Sources/BottomSheetPresentationController.swift

```diff
@@ -170,13 +203,22 @@ extension BottomSheetPresentationController: UIViewControllerInteractiveTransiti
 
 // MARK: - BottomSheetViewPresenterDelegate
 
-extension BottomSheetPresentationController: BottomSheetViewDelegate {
-    func bottomSheetViewDidTapDimView(_ view: BottomSheetView) {
-        dismiss(with: .zero)
-    }
+extension BottomSheetPresentationController: BottomSheetViewDismissalDelegate {
+    func bottomSheetView(_ view: BottomSheetView, willDismissBy action: BottomSheetView.DismissAction) {
+        guard presentationDelegate?.bottomSheetPresentationController(self, shouldDismissBy: action) ?? true else {
+            //view.reset()
+            presentationDelegate?.bottomSheetPresentationController(self, didCancelDismissBy: action)
```

Comment:
Do we really need this method? I feel like any code that would go into this delegate method could be in the `shouldDismissBy` method, except then it would be like a `willCancelDismissBy` instead.

Reply:
I agree it's more convenient to have a single method for this, but at the same time `shouldDismissBy` returns a boolean value, so in terms of good API design it shouldn't have any side-effects, like presenting alerts or whatsoever.